Library/Homebrew/os/mac/ruby_mach.rb

```diff
@@ -41,7 +41,14 @@ def mach_data
       end
 
       mach_data
-    rescue
+    rescue MachO::MagicError
+      # Silently ignore unsupported magic value (not a Mach-O binary) ...
+      []
+    rescue Exception => e
+      # ... but complain about other read errors for further investigation.
+      if ARGV.homebrew_developer?
```

Comment:
I guess ultimately it will have to become `ARGV.debug?` anyway and I'm not particularly attached to either variant. I'll change it in the next update. Thanks for the feedback!